packages/common/src/components/popover/Popover.vue

```diff
@@ -22,10 +23,38 @@ export default defineComponent({
     const isOpen = ref(false);
     const root = ref<HTMLElement | null>(null);
 
-    const toggle = () => (isOpen.value = !isOpen.value);
-    const open = () => (isOpen.value = true);
+    const toggle = () => (
+      (isOpen.value = !isOpen.value), nextTick(() => _handlePopoverPosition())
+    );
+    const open = () => (
+      (isOpen.value = true), nextTick(() => _handlePopoverPosition())
+    );
     const close = () => (isOpen.value = false);
 
+    const _handlePopoverPosition = () => {
+      if (!isOpen.value) {
+        return;
+      }
+      const popover = document.getElementById('popover');
+      const placeholderRect = root.value?.getBoundingClientRect();
+      const dropdownRect = popover?.getBoundingClientRect();
+
+      const dropdownRightX = dropdownRect.x + dropdownRect.width;
+      const placeholderRightX = placeholderRect.x + placeholderRect.width;
+
+      if (dropdownRect.x < 0) {
+        popover.style.left = '0';
```

Comment:
Would it be better to bind the inline styles using Vue's binding?

https://vuejs.org/v2/guide/class-and-style.html#Object-Syntax-1